flutter/lib/src/native/java/android_replay_recorder.dart

```diff
@@ -48,3 +73,141 @@ class AndroidReplayRecorder extends ScheduledScreenshotRecorder {
     }
   }
 }
+
+// Based on https://dart.dev/language/isolates#robust-ports-example
+class _AndroidNativeReplayWorker {
+  final SendPort _commands;
+  final ReceivePort _responses;
+  final Map<int, Completer<Object?>> _activeRequests = {};
+  int _idCounter = 0;
+  bool _closed = false;
+
+  static Future<_AndroidNativeReplayWorker> spawn() async {
+    // Create a receive port and add its initial message handler
+    final initPort = RawReceivePort();
+    final connection = Completer<(ReceivePort, SendPort)>.sync();
+    initPort.handler = (SendPort commandPort) {
+      connection.complete((
+        ReceivePort.fromRawReceivePort(initPort),
+        commandPort,
+      ));
+    };
+
+    // Spawn the isolate.
+    try {
+      await Isolate.spawn(_startRemoteIsolate, (initPort.sendPort),
+          debugName: 'SentryReplayRecorder');
+    } on Object {
+      initPort.close();
+      rethrow;
+    }
+
+    final (ReceivePort receivePort, SendPort sendPort) =
+        await connection.future;
+
+    return _AndroidNativeReplayWorker._(receivePort, sendPort);
+  }
+
+  _AndroidNativeReplayWorker._(this._responses, this._commands) {
+    _responses.listen(_handleResponsesFromIsolate);
+  }
+
+  Future<Object?> nativeAddScreenshot(_WorkItem item) async {
+    if (_closed) throw StateError('Closed');
```

Comment:
Note that this code path will not release the `jBuffer` eagerly, which is fine if we're throwing the state error to crash.

Reply:
thx. I've moved jbuffer creation to the isolate so this is resolved now